broadcaster/backlog/backlog.go

```diff
@@ -56,24 +56,30 @@ func (b *backlog) Head() BacklogSegment {
 	return b.head.Load()
 }
 
-func (b *backlog) backlogSizeInBytes() uint64 {
+func (b *backlog) backlogSizeInBytes() (uint64, error) {
 	if b.head.Load() == nil || b.tail.Load() == nil {
-		return 0
+		return 0, errors.New("the head or tail segment of feed backlog is nil")
```

Comment:
I think this should be `0, nil` because if the head and tail are nil then there's no issue but also no backlog

Reply:
if both are nil then its not a problem, but if either one of them is then its an error correct? I can change the OR to XOR